test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -802,6 +802,500 @@ test.describe( 'Navigation block', () => {
+			await requestUtils.createNavigationMenu( {
+				title: 'Hidden menu',
+				content: `
+					<!-- wp:navigation-link {"label":"Item 1","type":"custom","url":"http://www.wordpress.org/","isTopLevelLink":true} /-->
+					<!-- wp:navigation-link {"label":"Item 2","type":"custom","url":"http://www.wordpress.org/","isTopLevelLink":true} /-->
+					`,
+			} );
+			await editor.insertBlock( {
+				name: 'core/navigation',
+				attributes: { overlayMenu: 'always' },
+			} );
+			await editor.saveSiteEditorEntities();
+			await page.goto( '/' );
+		} );
+
+		test( 'overlay menu opens on click on open menu button', async ( {
+			page,
+		} ) => {
+			const overlayMenu = page.locator(
+				'nav.wp-block-navigation div.has-modal-open'
+			);
+			const openMenuButton = page.locator( '[aria-label="Open menu"]' );
+			await expect( overlayMenu ).toBeHidden();
+			await openMenuButton.click();
+			await expect( overlayMenu ).toBeVisible();
+		} );
+
+		test( 'overlay menu closes on click on close menu button', async ( {
+			page,
+		} ) => {
+			const overlayMenu = page.locator(
+				'nav.wp-block-navigation div.has-modal-open'
+			);
+			const openMenuButton = page.locator( '[aria-label="Open menu"]' );
+			const closeMenuButton = page.locator( '[aria-label="Close menu"]' );
+			await expect( overlayMenu ).toBeHidden();
+			await openMenuButton.click();
+			await expect( overlayMenu ).toBeVisible();
+			await closeMenuButton.click();
+			await expect( overlayMenu ).toBeHidden();
+		} );
+
+		test( 'overlay menu closes on ESC key', async ( { page } ) => {
+			const overlayMenu = page.locator(
+				'nav.wp-block-navigation div.has-modal-open'
+			);
+			const openMenuButton = page.locator( '[aria-label="Open menu"]' );
+			await expect( overlayMenu ).toBeHidden();
+			await openMenuButton.click();
+			await expect( overlayMenu ).toBeVisible();
+			await page.keyboard.press( 'Escape' );
+			await expect( overlayMenu ).toBeHidden();
+			await expect( openMenuButton ).toBeFocused();
+		} );
+
+		test( 'overlay menu focuses on first element after opening', async ( {
+			page,
+		} ) => {
+			const overlayMenu = page.locator(
+				'nav.wp-block-navigation div.has-modal-open'
+			);
+			const openMenuButton = page.locator( '[aria-label="Open menu"]' );
+			const firstItem = page.locator(
+				'ul li.wp-block-navigation-item:first-child > *:first-child'
+			);
+			await expect( overlayMenu ).toBeHidden();
+			await openMenuButton.click();
+			await expect( overlayMenu ).toBeVisible();
+			await expect( firstItem ).toBeFocused();
+		} );
+
+		test( 'overlay menu traps focus', async ( { page } ) => {
+			const overlayMenu = page.locator(
+				'nav.wp-block-navigation div.has-modal-open'
+			);
+			const openMenuButton = page.locator( '[aria-label="Open menu"]' );
+			const firstItem = page.locator(
+				'ul li.wp-block-navigation-item:first-child > *:first-child'
+			);
+			const closeMenuButton = page.locator( '[aria-label="Close menu"]' );
```

Comment:
Here, it would be valuable to put the focus and use only keyboard for interactions.